dht/token-sharding.hh

```diff
@@ -112,3 +126,21 @@ struct fmt::formatter<dht::sharder> {
                               sharder.shard_count(), sharder.sharding_ignore_msb());
     }
 };
+
+template<>
+struct fmt::formatter<dht::shard_replica_set> : fmt::formatter<std::string_view> {
+    template <typename FormatContext>
+    auto format(const dht::shard_replica_set& rs, FormatContext& ctx) const {
+        return fmt::format_to(ctx.out(), "{{{}}}", fmt::join(rs, ", "));
+    }
+};
+
+namespace std {
+
+inline
+std::ostream& operator<<(std::ostream& out, const dht::shard_replica_set& rs) {
+    fmt::print(out, "{}", rs);
+    return out;
+}
```

Comment:
Please use fmt exclusively, the automatic fallback to ostream doesn't work with fmt 10.

Reply:
wait, you fall back from ostream to fmt here. Why?

This is either recursive or unnecessary.

Reply:
boost macros (REQUIRE_EQUAL) require operator<<.

Why recursive? operator<< falls back to fmt, fmt formats directly.

Reply:
If there is no fmt support for shard_replica_set, then it will format through the ostream fallback, which is this function.

If there is fmt support (should be, via fmt/ranges.h), then this is unneeded and should be removed.

Reply:
I dropped the operator<<, looks like the code compiles without it now. I don't know what changed, but before it didn't.